pyccel/ast/functionalexpr.py

```diff
@@ -56,9 +56,16 @@ class FunctionalFor(TypedAstNode):
     target_type : PyccelSymbol, optional
         The type of the result of the functional for. This is useful at
         the syntactic stage to pass along the final type of the lhs (list/set/array/etc).
+    conditions : list[If|None], optional
+        A list of filter conditions corresponding to each for-loop in the comprehension.
+        Each element of this list is either an `If` instance that describes the filtering
+        condition for that loop, or `None` if no condition is applied in that loop.
```

Comment:
Does this fix #1410 ? It should be mentioned in the CHANGELOG

Reply:
Yes, it does. Added!

Reply:
What happens if `conditions` is not provided? Should the code convert this to `[None]`?